java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -2845,21 +3050,147 @@ protected boolean cleanImpl(boolean logErrorIfNotClean) {
 
     @Override
     public boolean isClean() {
-      return viewHandle == 0 && columnHandle == 0 && data == null && valid == null && offsets == null;
+      return viewHandle == 0 && columnHandle == 0 && toClose.isEmpty();
     }
 
     /**
      * This returns total memory allocated in device for the ColumnVector.
      * @return number of device bytes allocated for this column
      */
     public long getDeviceMemorySize() {
+      BaseDeviceMemoryBuffer valid = getValid();
+      BaseDeviceMemoryBuffer data = getData();
+      BaseDeviceMemoryBuffer offsets = getOffsets();
       long size = valid != null ? valid.getLength() : 0;
       size += offsets != null ? offsets.getLength() : 0;
       size += data != null ? data.getLength() : 0;
       return size;
     }
   }
 
+  public static ColumnVector createNestedColumnVector(HostMemoryBuffer data, List<HostMemoryBuffer> valid,
+      List<HostMemoryBuffer> offsets, List<DType> allTypes, List<Long> allRows, Optional<Long> nullCount) {
+    return NestedColumnVector.createColumnVector(data, valid, offsets, allTypes, allRows, nullCount);
+  }
+
+  private static class NestedColumnVector {
```

Comment:
Can we just make `copyToHost` and `copyToDevice` closer to what I put up above in the comments? I think you have the recursion here at the wrong level.

Reply:
Done.